### DISCUSSION TITLE
Is it possible to make dynamic alias resolution?

### DISCUSSION THREAD

**Discussion by denik1981** [webpack/webpack#14207] [Q&A]
To simplify the question say that I have this tree structure of files & folders
```
components
   componentA
      componentA.tsx
      other files for componentA(styles, types, etc)
   componentB
      componentB.tsx
      other files for componentB(styles, types, etc)
   ...
   componentZ
      componentZ.tsx
        other files for componentZ(styles, types, etc)
```

Now I would like to import a component without referenciating the folder and use instead a dynamic resolution where the name path is expanded to incorporate the folder name as well .

So instead of doing:
```
import ComponentA from 'components/componentA/componentA'
```
I would like to do:
``` 
import ComponentA from 'components/componentA'
```
But I don't want to write an alias for each folder inside the components tree. So I'm wondering if there is a way I can use a placeholder/wildcard/glob in the alias configuration of webpack so it can expand the last part of my path to include a folder with the same name. Is this possible? Preferrably without any third party plugin.

So in a very basic pseudo code I would expect something like this:
```
webpack: {
    alias: {
      '@components/&': path.resolve(__dirname, 'src/components/&/'),
    },
```
The & here should be later expanded to incorporate the last part of the path to resolve. And then I will use the "@components" to better trigger the alias resolution.

Is this possible?

**Comment by alexander-akait** [comment]
You can write function that collect paths and set aliases as you want
